cluster/src/main/java/org/apache/spark/streaming/JavaSnappyStreamingJob.java

```diff
@@ -47,7 +48,8 @@ final public Object runJob(Object sc, Config jobConfig) {
       SnappyUtils.setSessionDependencies(context.snappySession().sparkContext(),
           this.getClass().getCanonicalName(),
           Thread.currentThread().getContextClassLoader());
-      return runSnappyJob(context, SnappySessionFactory.cleanJobConfig(jobConfig));
+      return runSnappyJob(context, SnappySessionFactory.updateCredentials(context.snappySession()
+          , jobConfig));
```

Comment:
Why is this required here when code factory already sets it? I guess the fact that SnappyStreamingContext uses a singleton SnappySession is a problem in itself.

Reply:
This does two things: 1) Sets the credentials passed while submitting the job into the snappysession and 2) removes boot credentials from job config.

> Why is this required here when code factory already sets it?

You mean SnappySessionFactory? It does it only for SnappySQLJob and JavaSnappySQLJob

Reply:
Hmm, ok. But I still don't understand how this is supposed to work with streaming context. The SnappyStreamingContext is a singleton, and thus is its snappySession field. How setting credentials from multiple jobs into the same singleton SnappySession will work correctly? What does this mean in the first place? Only one user can submit streaming jobs, or all streaming related jobs should be done in only one job?

The whole old streamingContext model that can be started only once is a broken design to me and should perhaps be deprecated or taken out from our jobs given that the new streaming model is fairly decent now. Please discuss with the whole team.